youtube_dl/options.py

```diff
@@ -350,6 +350,10 @@ def _scrub_eq(o):
         '--video-password',
         dest='videopassword', metavar='PASSWORD',
         help='Video password (vimeo, smotri, youku)')
+    authentication.add_option(
+        '--auth-provider',
+        dest='auth_provider', metavar='AUTH_PROVIDER',
+        help='Authentication provider (nhl, rogers, ...)')
```

Comment:
If this auth provider thing is only related to nhl.com site then it should have `nhl` in it's name. Or if you suggest this to be a generic option to be used across extractors you have to make a complete documented proposal of how this is supposed to be used.